### PR TITLE
[ML] Adds support for a global calendars

### LINKED ISSUE #45013: [ML] Calendars : Allow calendar to apply to all jobs
Allow a ML calendar to apply to all ML jobs defined in the system.

Currently, calendars have lists of jobs and/or groups that the calendar applies to. There are some situations where the calendar should apply to all ML jobs. There is no mechanism for this. It is possible to put all jobs into a default group and attach that to the calendar, but there is an administrator overhead to making sure all new jobs are put in this group.

Some brief discussion in the forum: https://discuss.elastic.co/t/calendar-events-and-machine-learning-jobs/191678

### PR DESCRIPTION
**Pull request by benwtrent**
This adds `_all` to Calendar searches. This enables users to supply the `_all` string in the `job_ids` array when creating a Calendar. That calendar will now be applied to all jobs (existing and newly created).

Closes #45013

### CODE REVIEW

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Review by droberts195 — APPROVED** [review]
LGTM

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/CalendarQueryBuilder.java [hunk 1]
```diff
@@ -64,6 +65,7 @@ public SearchSourceBuilder build() {
         if (jobId != null) {
             jobIdAndGroups.add(jobId);
         }
+        jobIdAndGroups.add(MetaData.ALL);
```
Comment (benwtrent) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/CalendarQueryBuilder.java]: this should be moved under the `if (jobIdAndGroups.isEmpty() == false) {` clause.

**Comment by benwtrent** [comment]
Currently blocked by: https://github.com/elastic/kibana/issues/53604

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Pull request merged**